### PR TITLE
Feature: Pet rarity on pet drop message

### PR DESCRIPTION
**Pull request by ItsEmpa**
## What
Adds the rarity of the dropped pet to its drop chat message. 

https://ptb.discord.com/channels/997079228510117908/1204000873332871228

Example

## Changelog New Features
+ Added rarity text to pet drop messages. - Empa

### CODE REVIEW

**Comment by jani270** [comment]
> This pull request has conflicts with the base branch "beta". Please resolve those so we can test out your changes.

lol

**Comment by ItsEmpa** [comment]
oops

File: src/main/java/at/hannibal2/skyhanni/config/features/chat/ChatConfig.java [hunk 1]
```diff
@@ -136,4 +136,14 @@ public String toString() {
     @ConfigEditorBoolean
     @FeatureToggle
     public boolean hideSkyMall = true;
+
+    @Expose
+    @ConfigOption(
+        name = "Pet Drop Rarity",
+        desc = "Shows what rarity the pet drop is in the pet drop message.\n" +
+            "§6§lPET DROP! §r§5§lEPIC §r§5Slug §6(§6+1300☘)"
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/config/features/chat/ChatConfig.java]: Those reset characters aren't needed (a color also resets previous color + formatting)
```suggestion
            "§6§lPET DROP! §5§lEPIC §5Slug §6(§6+1300☘)"
```

File: src/main/java/at/hannibal2/skyhanni/features/chat/RareDropMessages.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class RareDropMessages {
+
+    private val petDropPattern by RepoPattern.pattern(
+        "pet.petdropmessage",
+        "(?<message1>§6§lPET DROP!|§5§lGREAT CATCH! §r§bYou found a §r§7\\[Lvl 1]) (?:§r)?§(?<rarityColor>.)(?<petName>[^§(.]+)(?<message2>.*)"
+    )
+
+    val config get() = SkyHanniMod.feature.chat.petRarityDropMessage
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/chat/RareDropMessages.kt]: can be private

File: src/main/java/at/hannibal2/skyhanni/features/chat/RareDropMessages.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (!config) return;
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/chat/RareDropMessages.kt]: remove semicolon

File: src/main/java/at/hannibal2/skyhanni/features/chat/RareDropMessages.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.util.ChatComponentText
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class RareDropMessages {
+
+    private val petDropPattern by RepoPattern.pattern(
+        "pet.petdropmessage",
+        "(?<message1>§6§lPET DROP!|§5§lGREAT CATCH! §r§bYou found a §r§7\\[Lvl 1]) (?:§r)?§(?<rarityColor>.)(?<petName>[^§(.]+)(?<message2>.*)"
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/chat/RareDropMessages.kt]: better names for message1 and message2, maybe typeOfDrop and magicFind?

File: src/main/java/at/hannibal2/skyhanni/features/chat/RareDropMessages.kt [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        var petName = ""
+        var message1 = ""
+        var message2 = ""
+
+        petDropPattern.matchMatcher(event.message) {
+            message1 = group("message1")
+            rarityColor = group("rarityColor")
+            petName = group("petName")
+            message2 = group("message2")
+        } ?: return
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/chat/RareDropMessages.kt]: Join declaration and assignment for those variables, and move the creation of the new chat component also inside the matchMatcher block.

File: src/main/java/at/hannibal2/skyhanni/features/chat/RareDropMessages.kt [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+        event.chatComponent = ChatComponentText(
+            "$message1 §r§$rarityColor§l${colorCodeToRarity(rarityColor.first()).uppercase()} §r§$rarityColor$petName$message2"
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/chat/RareDropMessages.kt]: Remove the two `§r`

**Review by hannibal002 — CHANGES_REQUESTED** [review]
Have not tested, just code cleanup.
Please provide us with an example Hypixel message so we can test this feature.

**Comment by ItsEmpa** [comment]
done. some examples of hypixel messages are
```
§5§lGREAT CATCH! §r§bYou found a §r§7[Lvl 1] §r§aGuardian§r§b.
§6§lPET DROP! §r§5Baby Yeti §r§b(+§r§b168% §r§b✯ Magic Find§r§b)
§6§lPET DROP! §r§5Slug §6(§6+1300☘)
```
also made sure `§6§lPET DROP! §r§5Baby Yeti` worked in case hypixel sends a message without any magic find/farming fortune in them

**Review by hannibal002 — CHANGES_REQUESTED** [review]
Tested with the test strings. slug and baby yeti worked fine, just guardian didnt work

<details>
<summary>Image</summary>

image

</details>


Furthermore, not sure why, but it looks like my test logic got called multiple times. Will fix this independently of this pr.

**Review by hannibal002 — APPROVED** [review]
Never mind. My testing logic was broken. It works all as expected!

<details>
<summary>Image</summary>

image

</details>

**Pull request merged**